### DISCUSSION TITLE
Manually register component at slug

### DISCUSSION THREAD

**Discussion by yiliansource** [facebook/docusaurus#4825] [Q&A] [answered]
Is it possible to register a component at a specific slug manually, instead of having to specify it via the `pages/*` folder structure?

The usecase is that I want to group redirects from old pages to new pages in a single file, to prevent cluttering up my hierarchy with page files containing a single redirect.

I'm planning to implement the actual redirect using the [`<Redirect>` component](https://docusaurus.io/docs/docusaurus-core#redirect).

**Comment by slorber** [comment, marked as answer]
You'd rather configure redirects on the server, using your hosting provider configuration, cf what we do on the Docusaurus site itself using Netlify `_redirects`: https://github.com/facebook/docusaurus/blob/master/website/static/_redirects

Using client-side redirects like `<Redirect>` is not really the recommended way to redirect an app: the redirect is done very late, only once the React app has been mounted/hydrated on the client. Eventually, using our client redirects plugin will be better than `<Redirect>` because it creates simple html files, not requiring hydrating React.

> Is it possible to register a component at a specific slug manually, instead of having to specify it via the pages/* folder structure?




---

> Is it possible to register a component at a specific slug manually, instead of having to specify it via the pages/* folder structure?

This is not documented very well but this is what our plugins lifecycle allows (quite similarly to Gatsby's createPage)

https://docusaurus.io/docs/lifecycle-apis#async-contentloadedcontent-actions

```js
export default function myRedirectRoutesPlugin(context, options) {
  return {
    name: 'myRedirectRoutesPlugin',
    async contentLoaded({content, actions}) {

      actions.addRoute({
        path: '/myOldRoute',
        exact: true,
        component: '@site/src/components/MyCustomRedirectComponent.js',
      });
      actions.addRoute({
        path: '/myOldRoute2',
        exact: true,
        component: '@site/src/components/MyCustomRedirectComponent.js',
      });
    },
  };
}
```

But I don't feel it's the ideal solution for your usecase

**Reply by yiliansource** [reply]
I see, it's not ideal, but this is already great to know for plugin development.

Server-side redirects are not really an option for me at the moment, since I'm hosting using GitHub Pages. But I agree that those would be way better than client-side redirects.

Thanks for your answer nevertheless!

**Reply by yiliansource** [reply]
@slorber just stumbled across [plugin-client-redirects](https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-client-redirects), could this be what I was looking for?

**Reply by slorber** [reply]
@yiliansource maybe, but in all case you'd rather use server redirects instead of client redirects
